### DISCUSSION TITLE
How to render final output

### DISCUSSION THREAD

**Discussion by chanpod** [GrapesJS/grapesjs#4276] [Q&A] [answered]
So I have the editor working and storing/loading to/from my api. But I can't seem to figure out how to get the compiled output similar to the preview window in the pre-built grapesjs-preset-webpage.

I'm using angular and we aren't wanting to use the CMS for the entire page, just some sub-sections of a particular page. What is the command to render the final output of the CMS, NOT the editor. Just the raw html/css compiled together nicely in the iframe or whatever it's supposed to be in. Once again, just like the preview minus the little preview button still floating around.

Note* I've tried editor.getHtml(), but this is missing the styling. I've scoured the documentation and I just don't see it. I know I'm glancing over it but I'm at a loss. Been a couple hours and this seems like something simple. I know I'm just missing it.

**Comment by AntoineGGG** [comment]
Hey @chanpod, I don't know Angular but with React you can do this:

![image](https://user-images.githubusercontent.com/22123286/164631942-17a6d139-7b86-432c-b54c-7bfbf6c2d12a.png)
****

**Comment by artf** [comment]
There is [`editor.getCss()`](https://grapesjs.com/docs/api/editor.html#getcss) if you need the current CSS in the editor page

**Reply by chanpod** [reply]
I tried that but it still didn't look right. Hmm. Wonder if angular was doing it wrong so the styles didn't get picked up correctly. I'll give it another shot.

**Reply by chanpod** [reply]
Yeah, I did that and it's still not right. I'm super lost on how the CSS is supposed to be managed. So as mentioned, I'm using edit.getHtml() and edit.getCss() and putting that into the page. The output doesn't match what I see in the editor. Anything else you can point me to look at?

First image is no editor enabled at all.
Second is editor preview mode
Third image is full editor

If I could just get the second image minus the exit preview mdoe button, that'd be perfect. So is the function that's doing that magic available somewhere?

![grapesEditorOff](https://user-images.githubusercontent.com/4175422/164951688-ea0d29fc-e4bc-4ec4-8964-54020f54b9cf.png)
![previewGrapes(r)](https://user-images.githubusercontent.com/4175422/164951689-3ed18bbd-85a7-439d-bc1b-3795eebb6f80.png)
![editorGrapes](https://user-images.githubusercontent.com/4175422/164951690-d846686f-faba-4bce-83e9-0d24f3c5b9a5.png)

**Reply by artf** [reply]
If you're using external styles (eg. `config.canvas.styles`), those are not included in the export.

**Reply by chanpod** [reply]
ooh, does that plugin use external styles I wonder? If so, how do I get them?

**Reply by chanpod** [reply]
It's angular messing it up. It's sanatizing things so the innerHtml stuff isn't properly getting the styles applied. Sigh. Thanks for the help

**Comment by chanpod** [comment, marked as answer]
Angular sanatizes the css. Ended up shoving html/css into an iframe to scope it so the page css and editor css don't conflict.
